lib/internal/process/per_thread.js

```diff
@@ -213,24 +212,27 @@ function setupUncaughtExceptionCapture(exceptionHandlerState,
   // shouldAbortOnUncaughtToggle is a typed array for faster
   // communication with JS.
 
-  process.setUncaughtExceptionCaptureCallback = function(fn) {
+  process.setUncaughtExceptionCaptureCallback = function(owner, fn) {
     if (fn === null) {
-      exceptionHandlerState.captureFn = fn;
+      exceptionHandlerState.captureFns.delete(owner);
       shouldAbortOnUncaughtToggle[0] = 1;
       return;
     }
+
     if (typeof fn !== 'function') {
       throw new ERR_INVALID_ARG_TYPE('fn', ['Function', 'null'], fn);
     }
-    if (exceptionHandlerState.captureFn !== null) {
-      throw new ERR_UNCAUGHT_EXCEPTION_CAPTURE_ALREADY_SET();
+
+    if (typeof owner !== 'symbol') {
+      throw new ERR_INVALID_ARG_TYPE('owner', ['Symbol'], owner);
```

Comment:
I guess this makes this semver-major?

Reply:
why not use a Set and just add/remove the functions themselves

Reply:
@devsnek's suggestion seems like a good alternative, I'll try that 👍